SmartcarAuth/SmartcarAuth.swift

```diff
@@ -126,9 +128,24 @@ Smartcar Authentication SDK for iOS written in Swift 3.
             }
         }
 
-        if let singleSelectValue = singleSelect {
-            let singleSelectBoolValue = singleSelectValue.boolValue;
-            queryItems.append(URLQueryItem(name: "single_select", value: singleSelectBoolValue ? "true" : "false"))
+        var singleSelectAdded = false;
+
+        if let singleSelectObject = singleSelectOptions {
+            if let singleSelectVIN = singleSelectObject.vin {
+                queryItems.append(URLQueryItem(name: "single_select_vin", value: singleSelectVIN))
+                singleSelectAdded = true;
+            }
+            if singleSelectAdded == false {
+                queryItems.append(URLQueryItem(name: "single_select", value: "false"))
+                singleSelectAdded = true;
+            }
+        }
+
+        if singleSelectAdded == false {
```

Comment:
There is a possibility to add to the queryItems twice right? If you pass in singleSelectOptions without vin, single_select will be added twice?

Reply:
single_select wouldn't be added twice because the flag is set to true after I added `single_select=false` for objects without keys or with junk keys, but it wouldn't add `single_select=true` if you had something like 
```
single_select = true
single_select_options = VehicleInfo(pizza: '123i12738')
```
Where you would expect single_select to be true then

This is addressed!